fe-nc-news-project/src/Comments.jsx

```diff
@@ -1,34 +1,75 @@
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
-import { fetchCommentsByArticleId } from "./utils";
+import { fetchCommentsByArticleId, postComment } from "./utils";
 import Article from "./Article";
 
-export default function Comments() {
+export default function Comments({username}) {
   const [comments, setComments] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [commentText, setCommentText] = useState("");
+  const [posting, setPosting] = useState(false);
+  const [posted, setPosted] = useState(false);
   const { article_id } = useParams();
 
+  const id = article_id
+
   useEffect(() => {
     fetchCommentsByArticleId(article_id).then((result) => {
       setComments(result.data.comments);
-      setIsLoading(false)
+      setIsLoading(false);
     });
-  }, []);
+  }, [comments]);
+
+
+  function submitComment(e) {
+    e.preventDefault()
+    setPosting(true)
+    const commentToPost = {}
+    commentToPost.author = username
+    commentToPost.body = commentText
+    postComment(article_id, commentToPost).then((result) => {
+      const justPosted = result.data.comment
+      const newCommentArr = [justPosted[0], ...comments]
+      setComments(newCommentArr.reverse())
+      setPosting(false)
+      setPosted(true)
+      setTimeout(() => {
```

Comment:
Would be interested to know what the setTimeout is for - but I trust that this is needed for you to have included it in the first place!